java-security/src/main/java/com/sap/cloud/security/adapter/spring/SAPOfflineTokenServicesCloud.java

```diff
@@ -45,8 +51,20 @@
  * }
  * </pre>
  * 
- * By default it used Apache Rest Client for communicating with the OAuth2
- * Server.
+ * By default it used Apache Rest Client for communicating with the OAuth2 Server.<br>
+ *
+ * Spring Security framework initializes the {@link org.springframework.security.core.context.SecurityContext}
+ * with the {@code OAuth2Authentication} which is provided as part of {@link #loadAuthentication} method. <br>
+ * This gives you the following options:
+ * <li>
+ *     <ul>All Spring security features are supported that uses
+ *     {@link org.springframework.security.core.context.SecurityContext#getAuthentication()}</ul>
+ *     <ul>You can access the {@code Authentication} via {@link SecurityContextHolder#getContext()}
+ *     also within asynchronous threads.</ul>
+ *     <ul>You can access the {@code Token} via {@link SpringSecurityContext#getToken()}
+ *     also within asynchronous threads.</ul>
+ * </li>
+ *
  */
 public class SAPOfflineTokenServicesCloud implements ResourceServerTokenServices, InitializingBean {
 
```

Comment:
The part below is changed in the build pack integration PR, which will lead to merge conflicts.

Reply:
yes, I'm aware of.
I've solved the conflict in the buildackintegration branch.